### PR TITLE
Social: Create the Jetpack Social version of the Publicize panel

### PR DESCRIPTION
**Pull request by pablinos**
#### Changes proposed in this Pull Request:

Jetpack Social needs its own implementation of the Publicize panel,
displayed in the editor sidebar. This is because we're not handling paid
plans yet, and the paid features (Republicize) aren't available. When we
do, it's unlikely that we'll handle everything in the same way as the
Jetpack plugin.

The dependent PRs - #24464, #24408 and #24324 - are merged, so the panel is now fully operational!

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
- Build the Jetpack Social plugin
- Activate it and connect
- Open the block editor and observe that you have the Publicize sharing panel in the sidebar.
- You can enable and disable connections, and Publicize the post on publish, with an optional custom message

image

### CODE REVIEW

File: projects/plugins/social/src/js/components/panel/description.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * @param {object} props                        - The component properties.
+ * @param {boolean} props.isPostPublished       - Whether the post has been published.
+ * @param {boolean} props.isPublicizeEnabled    - Whether the main publicize toggle is enabled.
+ * @param {boolean} props.hasConnections        - Whether we have any Publicize connections.
+ * @param {boolean} props.hasEnabledConnections - Whether any connections are enabled.
+ * @returns {object} The description component.
+ */
+export default function Description( props ) {
+	const descrip = getDescription( props );
+	return <div>{ descrip }</div>;
```
Comment (danielpost) [inline projects/plugins/social/src/js/components/panel/description.jsx:54]:
```suggestion
	return <div>{ getDescription( props ) }</div>;
```
For some reason I really don't like the `descrip` abbreviation 😆

File: projects/plugins/social/src/js/components/panel/index.jsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+			}
+
+			refresh();
+		},
+		[ hasEnabledConnections, refresh ]
+	);
+
+	// Panel wrapper.
+	const PanelWrapper = prePublish ? Fragment : PanelBody;
+	const wrapperProps = prePublish ? {} : { title: __( 'Share this post', 'jetpack-social' ) };
```
Comment (danielpost) [inline projects/plugins/social/src/js/components/panel/index.jsx:49]: Clever, I like it!

**Review by danielpost — DISMISSED** [review]
Awesome! Let's get it merged.

File: projects/plugins/social/src/js/components/panel/description.jsx (hunk 1 above)
Comment (danielpost) [reply projects/plugins/social/src/js/components/panel/description.jsx:54]: The entire function could probably be simplified to this if we wanted to:

```js
const Description = props => <div>{ getDescription( props ) }</div>;

export default Description;
```

File: projects/plugins/social/src/js/components/panel/index.jsx (hunk 2 above)
Comment (pablinos) [reply projects/plugins/social/src/js/components/panel/index.jsx:49]: I can't take the credit for that, it's in the Jetpack version too. I think that might have been @retrofox's handywork.

File: projects/plugins/social/src/js/components/panel/description.jsx (hunk 1 above)
Comment (pablinos) [reply projects/plugins/social/src/js/components/panel/description.jsx:54]: Yes good point, I think this is hangover from when I was doing more in that component. I'll get that changed.

Comment (pablinos) [reply projects/plugins/social/src/js/components/panel/description.jsx]: Addressed in ac1357a

**Review by danielpost — APPROVED** [review]

**Comment by anomiex** [comment]
r246564-wpcom

**Pull request merged**